relay-pattern/src/typed.rs

```diff
@@ -141,6 +143,85 @@ impl<C: PatternConfig> TypedPatterns<C> {
     }
 }
 
+impl<C: PatternConfig> Default for TypedPatterns<C> {
+    fn default() -> Self {
+        Self::builder().build()
+    }
+}
+
+impl<C> PartialEq for TypedPatterns<C> {
+    fn eq(&self, other: &Self) -> bool {
+        self.raw.eq(&other.raw)
+    }
+}
+
+impl<C: PatternConfig> From<String> for TypedPatterns<C> {
+    fn from(value: String) -> Self {
+        [value].into_iter().collect()
+    }
+}
+
+impl<C: PatternConfig> From<Vec<String>> for TypedPatterns<C> {
+    fn from(value: Vec<String>) -> Self {
+        value.into_iter().collect()
+    }
+}
+
+impl<C: PatternConfig, const N: usize> From<[String; N]> for TypedPatterns<C> {
+    fn from(value: [String; N]) -> Self {
+        value.into_iter().collect()
+    }
+}
+
+/// Creates [`Patterns`] from an iterator of strings.
+///
+/// Invalid patterns are ignored.
+impl<C: PatternConfig> FromIterator<String> for TypedPatterns<C> {
+    fn from_iter<T: IntoIterator<Item = String>>(iter: T) -> Self {
+        let mut builder = Self::builder();
+        for pattern in iter.into_iter() {
+            let _ = builder.add(pattern);
+        }
+        builder.build()
+    }
+}
+
+impl<C> From<TypedPatterns<C>> for Patterns {
+    fn from(value: TypedPatterns<C>) -> Self {
+        value.patterns
+    }
+}
+
+impl<C> AsRef<Patterns> for TypedPatterns<C> {
+    fn as_ref(&self) -> &Patterns {
+        &self.patterns
+    }
+}
+
+impl<C> Deref for TypedPatterns<C> {
+    type Target = Patterns;
+
+    fn deref(&self) -> &Self::Target {
+        &self.patterns
+    }
+}
+
+impl<C> Clone for TypedPatterns<C> {
```

Comment:
Since `PhantomData` is `Clone` do we need this `Clone` impl?

Reply:
Yes, because the derive creates a clone impl which has all of the generics in its bounds, but we can impl clone independently of the generics.